src/Atmos/Model/ref_state.jl

```diff
@@ -42,48 +42,60 @@ state.
 struct HydrostaticState{P, FT} <: ReferenceState
     virtual_temperature_profile::P
     relative_humidity::FT
+    subtract_off::Bool
 end
 function HydrostaticState(
-    virtual_temperature_profile::TemperatureProfile{FT},
+    virtual_temperature_profile::TemperatureProfile{FT};
+    subtract_off = true,
 ) where {FT}
     return HydrostaticState{typeof(virtual_temperature_profile), FT}(
         virtual_temperature_profile,
         FT(0),
+        subtract_off,
     )
 end
 
 vars_state(m::HydrostaticState, ::Auxiliary, FT) =
     @vars(ρ::FT, p::FT, T::FT, ρe::FT, ρq_tot::FT, ρq_liq::FT, ρq_ice::FT)
 
 atmos_init_ref_state_pressure!(m, _...) = nothing
-function atmos_init_ref_state_pressure!(
+function atmos_init_ref_state_pT!(
     m::HydrostaticState{P, F},
     atmos::AtmosModel,
     aux::Vars,
     geom::LocalGeometry,
 ) where {P, F}
     z = altitude(atmos, aux)
-    _, p = m.virtual_temperature_profile(atmos.param_set, z)
+    T_virt, p = m.virtual_temperature_profile(atmos.param_set, z)
     aux.ref_state.p = p
+    aux.ref_state.T = T_virt
```

Comment:
@tapios might need some input here. Should we do the same thing we do below using the `air_temperature_from_ideal_gas_law` conversion?

Reply:
Where do we define ref_state, why do not we save rho in the ref state ?

Reply:
Setting `aux.ref_state.T = T_virt` in general seems wrong (because temperature generally is not equal to virtual temperature). There is no reason I can see not to store density in the ref state.

Reply:
I'm not quite sure what is going on in this code here. It seems convoluted to me. What we need to do is (1) specify virtual temperature `Tv(z)` and pressure `p(z)` as functions of altitude `z`; (2) compute density `rho(z)` from the discrete derivative (l. 85); (3) update the virtual temperature to be discretely consistent with pressure and density, i.e., `Tv = p/(R_d rho)`. (Atmosphere design doc sections 7.1.2 and 7.1.3.) (4) Temperature and specific humidity then follow by solving the nonlinear equation (7.8) in the design doc, as done by `temperature_and_humidity_given_TᵥρRH`. This last part is unaffected by any changes in vertical discretization.